packages/react-aria-components/src/Checkbox.tsx

```diff
@@ -240,7 +239,10 @@ function Checkbox(props: CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
       data-invalid={isInvalid || undefined}
       data-required={props.isRequired || undefined}>
       <VisuallyHidden elementType="span">
-        <input {...mergeProps(inputProps, focusProps)} ref={inputRef} />
+        <input
+          // data attribute should only go to the outermost dom
+          {...mergeProps(removeDataAttributes(inputProps), focusProps)}
```

Comment:
thanks for the pr!

i think this approach is fine but i think we want to match how we did it in other components like in this pr: https://github.com/adobe/react-spectrum/pull/5008. there's a [comment](https://github.com/adobe/react-spectrum/pull/5008#discussion_r1320225114) on that pr which mentions why

Reply:
ok, I will try that later :)

Reply:
Thanks for the review!